odata-csdl/14 Vocabulary and Annotation.md

```diff
@@ -1364,56 +1364,172 @@ Addresses/-1/Street
+segments follow the rules for path expressions targeting the corresponding
+child element.
+
+For annotations hosted by an entity set or a singleton, the path is evaluated
+starting at the entity set or singleton, i.e. an empty path resolves to the
+entity set or singleton, and non-empty paths MUST follow the rules for
+annotations targeting the declared entity type of the entity set or singleton.
+
+For annotations hosted by an entity type or complex type, the path is
+evaluated starting at the type, i.e. an empty path resolves to the type, and
+the first segment of a non-empty path MUST be a structural or navigation
+property of the type, a [type cast](#TypeCast), or a [term cast](#TermCast).
+
+For annotations hosted by an action, action import, function, function
+import, parameter, or return type, the first segment of the path MUST be the
+name of a parameter of the action or function or `$ReturnType`.
+
+For annotations hosted by a structural or navigation property, the path
+evaluation rules additionally depend upon how the annotation target is
+specified, as follows:
+
+- If the annotation uses targeting and the target path starts with
+  an entity container, or the annotation is directly or indirectly embedded in such an
+  annotation, the path is evaluated starting at the declared type of the
+  hosting property. An empty path resolves to the declared type of the
+  property, and non-empty paths MUST follow the rules for annotations
+  targeting the declared type of the property. If the type is primitive, the
+  first segment of a non-empty path MUST be a [type cast](#TypeCast) or a
+  [term cast](#TermCast).
+
+- If the annotation uses targeting and the target path does not
+  start with an entity container, or the annotation is directly or indirectly
+  embedded in such an annotation, the path is evaluated starting at the *outermost*
+  entity type or complex type named in the target path. This
+  allows e.g. specifying the value of an annotation on one property to be
+  calculated from values of other properties of the outermost type. An empty
+  path resolves to the outermost type, and the first segment of a non-empty
+  path MUST be a structural or navigation property of the outermost type, a
+  [type cast](#TypeCast), or a [term cast](#TermCast).
+
+- If the annotation is directly or indirectly embedded in the hosting
+  property, the path is evaluated starting at the directly enclosing type of
+  the hosting property. This allows e.g. specifying the value of an annotation
+  on one property to be calculated from values of other properties of the same
+  enclosing type. An empty path resolves to the enclosing type, and non-empty
+  paths MUST follow the rules for annotations targeting the directly enclosing
+  type.
+
+::: example
+Example ##ex: Annotations hosted by property B in various modes
+
+The annotations in the first block are treated according to the third bullet point:
```

Comment:
Can we rearrange the bullet points or the examples so that both have the same sequence?

And why not use a numbered list instead of bullet points?